src/main/webapp/app/shared/components/customTab/custom-tab.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { ButtonGroup, Button } from 'reactstrap';
+import classnames from 'classnames';
+
+import './custom-tab.scss';
+import { Translate } from 'react-jhipster';
+
+export interface ITabProps {
+  currentTab: string;
+  tabList: ITabInfo[];
+}
+
+export interface ITabInfo {
+  tabName: string;
+  tabTranslateKey: string;
+  tabRoute: string;
+}
+
+export class CustomTab extends React.Component<ITabProps, {}> {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      <div className="overflow-x-scroll tab-container my-2">
+        <ButtonGroup>
+          <TabItems itemsList={this.props.tabList} activeTab={this.props.currentTab} />
+        </ButtonGroup>
+      </div>
+    );
+  }
+}
+
+const navigateTo = (path: string) => () => {
+  window.location.href = path;
```

Comment:
` window.location.href` will trigger an entire browser refresh.

I suggest to use `history` from react-router-dom instead
```
        history.push('/auth/email/login');
```

you can get history object from 
1. the [RouteProps](https://reactrouter.com/web/api/Route/route-props)
2. [useHistory Hooks](https://reactrouter.com/web/api/Hooks/usehistory)
// since this is a hook, it only works in functional component
const history = useHistory();
3. [withRouter High Order Component](https://reactrouter.com/web/api/withRouter)
// for class component

Reply:
@kachun333 
help me to verify use of history in de5b9a5ee6b46993ddb0abce6ef8a3c926ed10f8
and resolve this if it is correct
:relaxed:

Reply:
there are 3 options for you to choose from. But why you chose to create another history object...

read `app.tsx` you will notice that we are using BrowserRouter and it internally maintains its own history object. So we should use THAT history object. 

read this carefully.
https://stackoverflow.com/questions/42701129/how-to-push-to-history-in-react-router-v4

if you insist to create another history object, please change from BrowserRouter to a normal Router.

Reply:
@kachun333  can verify this? c36c1b9461e2a228290bea83a60769aa967c0be2
I used Link instead of my own click handler

Reply:
this is good.